aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs

```diff
@@ -17,16 +17,16 @@ public SamplePageFilter(IConfiguration config)
 
         public void OnPageHandlerSelected(PageHandlerSelectedContext context)
         {
-            var key = _config.GetValue(typeof(string), "UserAgentID");
+            var key = _config["UserAgentID"];
             context.HttpContext.Request.Headers.TryGetValue("user-agent", out StringValues value);
             ProcessUserAgent.Write(context.ActionDescriptor.DisplayName,
                                    "SamplePageFilter.OnPageHandlerSelected",
-                                    value, key.ToString());
+                                   value, key.ToString());
         }
 
         public void OnPageHandlerExecuting(PageHandlerExecutingContext context)
         {
-            Debug.WriteLine("Global sync PageHandlerExecutingContext called.");
+            Debug.WriteLine("Global sync OnPageHandlerExecuting called.");
```

Comment:
I've been using a variant of
```

            MyDebug.Write(MethodBase.GetCurrentMethod(), context.HttpContext.Request.Path);

public static class MyDebug
    {
        public static void Write(MethodBase m, string path)
        {
            Debug.WriteLine(m.ReflectedType.Name + "." + m.Name + " " +
                path);
        }
    }
```

Not sure if it's worth using here.

Reply:
Yeah, it's nice to ensure that bad copy/pastes are avoided. Overall, I think I would use `ILogger` for these messages and take advantage of the automatic log category stuff for the context. Combining that with `[CallerMemberName]` would remove the need for reflection too.

Reply:
`ILogger ` is not available to every filter in this sample.

Reply:
Makes sense.